app/code/core/Mage/Core/Model/Layout/Validator.php

```diff
@@ -233,9 +233,9 @@ public function validateTemplatePath(array $templatePaths)
         /** @var Varien_Simplexml_Element $path */
         foreach ($templatePaths as $path) {
             if ($path->hasChildren()) {
-                $path = stripcslashes(trim((string) $path->children(), '"'));
+                $path = stripcslashes(trim((string)$path->children(), '"'));
             }
-            if (strpos($path, '..' . DS) !== false) {
+            if (strpos((string)$path, '..' . DS) !== false) {
```

Comment:
Is this better?
```suggestion
            $path = $path->hasChildren()
                ? stripcslashes(trim((string)$path->children(), '"'))
                : (string)$path;
            if (strpos($path, '..' . DS) !== false) {
```
 For me it's easier to read, `$path` is typecast to string.

Reply:
I'll commit @kiatng code but I'll convert the strpos to str_contains since we have the polyfills